### ISSUE TITLE
ibm_iam_trusted_profile_link, argument "namespace" is not required if cr_type is VSI

### ISSUE THREAD

**Issue by sasikeda** [IBM-Cloud/terraform-provider-ibm#3219]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* ibm_iam_trusted_profile_link

### Expected Behavior

Documentation states;
```
namespace - (Required, String) The compute resource namespace, only required if cr_type is IKS_SA or ROKS_SA.
```

If `cr_type = VSI`, then `namespace` argument is not required in the link.

### Actual Behavior

My snippet;

```
resource "ibm_iam_trusted_profile_link" "itself" {
   profile_id = var.trusted_profile_id
   cr_type    = "VSI"
   link {
     crn = var.target_vsi_crn
   }
   name = var.trusted_profile_link_name
 }
```

Error;
```
Error: Missing required argument
│
│   on resources/ibmcloud/security/iam/trusted_profile_link/trusted_profile_link.tf line 20, in resource "ibm_iam_trusted_profile_link" "itself":
│   20:   link {
│
│ The argument "namespace" is required, but no definition was found.
```

**Comment by hkantare** [comment]
@pooraniss-ibm Can you please look into this issue..We should mark namespace as optional argument

**Comment by pooraniss-ibm** [comment]
yes thats correct: namespace should only be marked as optional. here is the [PR](https://github.com/IBM-Cloud/terraform-provider-ibm/pull/3222)
